### PR TITLE
Update rx utils

### PR DESCRIPTION
**Pull request by raveclassic**
@Nedgeva Please take a look

### CODE REVIEW

File: packages/rx-utils/src/context2.utils.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	}
+}
+
+export interface Context<E, A> extends Reader<E, Observable<A>> {}
+
+const memo = memoOnce(eqShallow);
+export const contextInstance: Monad2<URI> & ProductLeft<URI> & ReaderM1<URIObservable> = {
+	URI,
+	...getReaderM(observable),
+	asks: f => memo(e => observable.of(f(e))),
```
Comment (sutarmin) [inline packages/rx-utils/src/context2.utils.ts]: Could you explain, why do we need `memo` here? I can't come up with cases where we would call `asks` several times sequentially with the same argument.

Comment (raveclassic) [reply packages/rx-utils/src/context2.utils.ts]: The most obvious:
```typescript
interface Env {
  a: number
}
const a = context.asks((e: E) => 'some heavy computation' + e.a + new Date()))
a({ a: 1, b: Math.random() })
a({ a: 1, b: Math.random() })
a({ a: 1, b: Math.random() })
a({ a: 1, b: Math.random() })
a({ a: 1, b: Math.random() })
```

Each context call will rerun the inner computation although dependencies didn't change. This leads to a massive performance regression. However if we tweak `asks` and `context.combine` with shallow memo this won't happen.

Comment (raveclassic) [reply packages/rx-utils/src/context2.utils.ts]: >  I can't come up with cases where we would call asks several times sequentially with the same argument.

Just pass context to combine several times (like in several parts of application you render the same container - combine the same context):
```typescript
declare const A: Context<{ a: number }, ReactComponent>
const A1 = context.combine(A, (A) => <div><A/></div>)
const A2 = context.combine(A, (A) => <div><A/></div>)
const B = context.combine(A1, A2, (A1, A2) => <div><A1/><A2/></div>)
```

File: packages/rx-utils/src/rd/live-data.utils.ts [hunk 2]
```diff
@@ -1,6 +1,4 @@
-import { Observable } from 'rxjs';
-import { RemoteData } from '@devexperts/remote-data-ts';
-import { Option } from 'fp-ts/lib/Option';
-
-export type LiveData<L, A> = Observable<RemoteData<L, A>>;
-export type LiveDataOption<L, A> = Observable<RemoteData<L, Option<A>>>;
+/**
+ * @deprecated User `LiveData` and `LiveDataOption` from the root of `rx-utils`
```
Comment (sutarmin) [inline packages/rx-utils/src/rd/live-data.utils.ts]: I believe `User` is a misprint

**Comment by sutarmin** [comment]
Don't you think that it makes sense to add `BREAKING CHANGE` part to commit messages? It would be easier to estimate the cost of migration for the other guys then

**Comment by raveclassic** [comment]
> Don't you think that it makes sense to add `BREAKING CHANGE` part to commit messages? It would be easier to estimate the cost of migration for the other guys then

But nothing's broken

**Pull request merged**